File: javascript/ironredis/sentinel.js
'use strict';
if (!(process.argv.length === 5 || process.argv.length === 6)) {
    console.log("Usage: sentinelhost sentinelport service password")
    process.exit(1)
}

const Redis = require("ioredis");

var sentinelHost = process.argv[2];
var sentinelPort = process.argv[3];
var service = process.argv[4];
var password = undefined;
if (process.argv.length === 6) {
    password = process.argv[5];
}

var redis = new Redis({
    sentinels: [{
        host: sentinelHost,
        port: sentinelPort
    }], //you can add more sentinels here
    name: service,
    password: password
});

redis.on("error", function (error) {
    console.error("Error in connection");
    console.error(error);
});

redis.set("foo", "bar", function (err, result) {
    if (err) {
        return console.error(err);
    }
    console.log("SET:", result);
});


redis.get("foo").then(function (result) {
    console.log("GET:", result);
});
//close connection
redis.quit()

File: python/redispy/sentinel.py
import redis
import redis.sentinel
import sys


def main():
    if not (len(sys.argv) == 4 or len(sys.argv) == 5):
        print("Usage: sentinelhost sentinelport service password")
        exit(1)
    sentinel_host = sys.argv[1]
    sentinel_port = sys.argv[2]
    service_name = sys.argv[3]

    password = None
    if len(sys.argv) == 5:
        password = sys.argv[4]

    # you can add more sentinels to the list
    sentinel_list = [(sentinel_host, sentinel_port)]
    s = redis.sentinel.Sentinel(sentinel_list, socket_timeout=1.0)
    r = s.master_for(service_name, socket_timeout=1.0, password=password)

    print("Set: {}".format(r.set("foo", "bar")))
    print("Get: {}".format(r.get("foo")))

    # close the connection pool
    r.connection_pool.disconnect()


if __name__ == "__main__":
    main()
